examples/solana_native/basic_nft/README.md

```diff
@@ -0,0 +1,423 @@
+
+```rust
+gem.data.color = color;
+gem.data.rarity = rarity;
+gem.data.short_description = short_description;
+gem.data.mint = *mint.info.key;
+gem.data.assoc_account = Some(*assoc_token_account.key);
+
+csl_spl_token::src::cpi::initialize_mint_2(for_initialize_mint_2, 0, *wallet.key, None)?;
+csl_spl_assoc_token::src::cpi::create(for_create)?;
+csl_spl_token::src::cpi::mint_to(for_mint_to, 1)?;
+csl_spl_token::src::cpi::set_authority(for_set_authority, 0, None)?;
+```
+
+#### Burn business logic
+
+Copy and paste the following code just below the comment line `// Implement your business logic here...` in the file `burn.rs`
+
+```rust
+gem.data.assoc_account = None;
+csl_spl_token::src::cpi::burn(for_burn, 1)?;
+```
+
+#### Transfer business logic
+
+Copy and paste the following code just below the comment line `// Implement your business logic here...` in the file `transfer.rs`
+
+```rust
+gem.data.assoc_account = Some(*destination.key);
+
+// Create the ATA account for new owner if it hasn't been created
+if assoc_token_account.lamports() == 0 {
+    csl_spl_assoc_token::src::cpi::create(for_create)?;
+}
+
+csl_spl_token::src::cpi::transfer_checked(for_transfer_checked, 1, 0)?;
+```
+
+### 4. Build and deploy the program
+
+Open the terminal and navigate to the `program` directory, from there execute the following command:
+
+```shell
+cargo build-sbf
+```
+If it gives an error then run the following command:
+```shell
+cd program/
+```
+**I think this will work**
+
+Run a local Solana validator by opening a new terminal and typing the command:
+
+> Don’t close this terminal because it is required for the following steps
+
+```shell
+solana-test-validator
+```
+
+Deploy the program by opening a new terminal and navigating to the `program` directory; from there, execute the following command:
+
+```shell
+solana program deploy target/deploy/counter.so
+```
+
+After deploying the program, you will receive the program id; copy and paste it somewhere for later.
+
+### 5. Test your contract
+
+Create a new file named `app.ts` inside the directory `program_client` and copy and paste the following code into the `app.ts` file:
+
+> Replace the “PASTE_YOUR_PROGRAM_ID” with the program id you got when deploying the Solana program.
+
+```typescript
+import {Connection, Keypair, PublicKey, sendAndConfirmTransaction, SystemProgram, Transaction,} from "@solana/web3.js";
+import * as fs from "fs/promises";
+import * as path from "path";
+import * as os from "os";
+import {
+    burnSendAndConfirm,
+    CslSplTokenPDAs,
+    deriveGemMetadataPDA,
+    getGemMetadata,
+    initializeClient,
+    mintSendAndConfirm,
+    transferSendAndConfirm,
+} from "./index";
+import {getMinimumBalanceForRentExemptAccount, getMint, TOKEN_PROGRAM_ID,} from "@solana/spl-token";
+
+async function main(feePayer: Keypair) {
+    const args = process.argv.slice(2);
+    const connection = new Connection("https://api.devnet.solana.com", {
```

Comment:
Change "https://api.devnet.solana.com" for "http://127.0.0.1:8899" since based on the above commands it will be deployed to this validator